File: Implementation/c++/04_보너스점수.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/17389
// 2

#include <iostream>

using namespace std;

int main()
{
	int N;
	string s;
	cin >> N >> s;

	int result = 0, bonus = 0;
	for (int i = 0; i < N; i++) {
		if (s[i] == 'O')
			result += i + 1 + bonus++;
		else
			bonus = 0;
	}
	cout << result;
}

File: backup/Implementation/python/04_보너스점수.py
# fast campus 강의
# https://www.acmicpc.net/problem/17389
# 2

N, S = input(), input()

score, bonus = 0, 0
for idx, OX in enumerate(S):
    if OX == 'O':
        score += idx + 1 + bonus
        bonus += 1
    else:
        bonus = 0

print(score)